pkgs/stdenv/generic/get-env.sh.nix

```diff
@@ -0,0 +1,136 @@
+builtins.toFile "get-env.sh" ''
+
+  set -e
+  if [ -e .attrs.sh ]; then source .attrs.sh; fi
+
+  export IN_NIX_SHELL=impure
+  export dontAddDisableDepTrack=1
+
+  if [[ -n $stdenv ]]; then
+      source $stdenv/setup
```

Comment:
This also reminds me that this doesn't actually make it be in sync with `mkDerivation`, because that sources `$stdenv/setup` separately [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/default-builder.sh#L1)

Ideally we should have a tight integration with `mkDerivation` and its phases, which seems pretty hard, but maybe doable since we're in nixpkgs now.